src/v/cluster/tm_stm_cache.h

```diff
@@ -236,13 +240,27 @@ class tm_stm_cache {
     std::deque<tm_transaction> checkpoint();
 
 private:
+    struct tx_wrapper {
+        tx_wrapper() {}
+
+        tx_wrapper(const tm_transaction& tx) { this->tx = tx; }
+
+        tm_transaction tx;
+        intrusive_list_hook _hook;
+    };
```

Comment:
any reason to not to the _hook in tm_transction? might simplify some things..

Reply:
- tm_transction is being saved to the log; I don't want to debug interaction of the new field, adl, serde etc.
- all tm_transction's fields have "protocol" meaning;  _hook is an ad-hoc property which makes sense only within the cache so by keeping it private we limit the scope of things a reader should understand